pkg/api/error.go

```diff
@@ -19,7 +19,17 @@ type CloudError struct {
 }
 
 func (err *CloudError) Error() string {
-	return fmt.Sprintf("%d: %s: %s: %s", err.StatusCode, err.Code, err.Target, err.Message)
+	var details string
+	if len(err.Details) > 0 {
+		details = " Details: "
+		for i, innerErr := range err.Details {
+			details += (&CloudError{0, &innerErr}).Error()
```

Comment:
By creating a CloudError here using the CloudErrorBody from Details, Error can be called recursively to get information from nested errors.